### PR TITLE
clazo/APPEALS-46256

### PR DESCRIPTION
**Pull request by chrislazo22**
Resolves [Update error handling for BIS calls](https://jira.devops.va.gov/browse/APPEALS-46256)

# Description
* Remove `raise` on failed BIS call
* Log error to EventAudit
* Set `event.error` to `nil` when event is processed 

## Acceptance Criteria
- [ ] All errors raised from BIS `fetch_veteran_info` calls will be handled. They will be logged and concatenated in `EventAudit` notes column
- [ ] All errors raised from BIS `fetch_person_info` calls will be handled. They will be logged and concatenated in `EventAudit` notes column
- [ ] All errors raised from BIS `fetch_limited_poas_by_claim_ids` calls will be handled. They will be logged and concatenated in `EventAudit` notes column
- [ ] All errors raised from BIS `fetch_rating_profiles_in_range` calls will be handled. They will be logged and concatenated in `EventAudit` notes column
- [ ]  The "error" column of an `Event` record will always be cleared when an `Event` is successfully processed.

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/APPEALS-47736)
2. [Test Xray Exec](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-48280&testIssueKey=APPEALS-47736)

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Backend

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- Any errors that arise from BIS will be handled by logging the error to the notes column on the EventAudit for traceability, but will not cause the job to fail.

### CODE REVIEW

File: app/jobs/base_event_processing_job.rb [hunk 1]
```diff
@@ -53,6 +53,7 @@ def start_processing!
   def complete_processing!
     ActiveRecord::Base.transaction do
       @event.processed!
+      @event.error = nil
```
Comment (Aaron-Willis) [inline app/jobs/base_event_processing_job.rb:56]: Please use a strong validation #update! here.

File: app/models/builders/decision_review_created/request_issue_builder.rb [hunk 2]
```diff
@@ -257,8 +257,11 @@ def assign_nonrating_issue_bgs_id
     @request_issue.nonrating_issue_bgs_id = issue.prior_non_rating_decision_id&.to_s
   end
 
-  def assign_nonrating_issue_bgs_source
-    @request_issue.nonrating_issue_bgs_source = issue.prior_decision_source&.to_s
+  def calculate_nonrating_issue_bgs_source
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_created/request_issue_builder.rb:260]: Please update this conditional to check if the following evaluates to true: 
 decision_review_created.ep_code_category == NONRATING_EP_CODE_CATEGORY.  If it does, then we will update the nonrating_issue_bgs_source with the prior_decision_source value.

File: spec/factories/decision_review_created.rb [hunk 3]
```diff
@@ -575,7 +575,7 @@
             "legacy_appeal_issue_id" => nil,
             "source_contention_id_for_remand" => nil,
             "source_claim_id_for_remand" => nil,
-            "prior_decision_source" => "CORP_AWARD_ATTORNEY_FEE"
+            "prior_decision_source" => "NON_RATING"
```
Comment (Aaron-Willis) [inline spec/factories/decision_review_created.rb:578]: "NON_RATING" would not be an expected value for prior_decision_source

File: app/models/builders/decision_review_created/request_issue_builder.rb [hunk 4]
```diff
@@ -257,8 +257,11 @@ def assign_nonrating_issue_bgs_id
...
+    @request_issue.nonrating_issue_bgs_source =
+      if decision_review_created.ep_code_category == NONRATING_EP_CODE_CATEGORY
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_created/request_issue_builder.rb:262]: Do we need to add a 'nil' assignment here?  or does ruby automatically assign it to nil?

Comment (chrislazo22) [reply app/models/builders/decision_review_created/request_issue_builder.rb:262]: `@request_issue.nonrating_issue_bgs_source` will be assigned `nil` if `decision_review_created.ep_code_category == NONRATING_EP_CODE_CATEGORY` is false

**Review by Aaron-Willis — APPROVED** [review]

**Pull request merged**